### ISSUE TITLE
Toxcore only processes File Transfers in series.

### ISSUE THREAD

**Issue by GrayHatter** [irungentoo/toxcore#1468]
Currently toxcore will only send files in series order. 

Toxcore should attempt to transfer files in parallel. 

This gives the option to the clients to either automagically pause transfers, and send them one by one, or to send X number at a time, or to send them all concurrently. With the way it is now clients aren't able to chose how they want files sent, and can only send them one by one. Possibly giving the impression that file transfers are "broken" simply because toxcore hasn't gotten to them yet.

**Comment by LittleVulpix** [comment]
This used to be the case pre-new tox api.

I would still prefer that filetransfers are initially made stable before multiple filetransfers at the same time are allowed. As it is right now I can't even get one file across. Now imagine my mom queues up 20 photos from her trip at once. It worked in Skype; drop and forget. In Tox, I'm 100% sure she's getting disconnected.

tl;dr - good idea, but filetransfers have to be stable first, AKA irungentoo/toxcore#1220

**Comment by fcore117** [comment]
+1 LittleVulpix, i would prefer single file sending too because main goal is rock stability and correct file arrival.

**Comment by aaannndddyyy** [comment]
i have no strong opinion on parallel versus sequential, though I'd prefer parallel too. But it's just a psychological factor, in the end, the overall transfer time is the same - and that only if the transfers do not have to restart from zero, like currently the case at least in qtox, after the inevitable disconnect.

**Comment by tux3** [comment]
@aaannndddyyy This is annoying when we factor in avatars and proposed features like stickers. Those currently have to wait until the end of any previous file transfer, and that can be a long time.

**Comment by TheNain38** [comment]
@tux3 +1

**Comment by aaannndddyyy** [comment]
@tux3 That's a good point. Though, I can imagine that if you really want real-time stickers implemented via file transfers, you'd also want prioritizing of file transfers - otherwise even with parallel transfers you could delay the stickers.

**Comment by jf99** [comment]
Transferring multiple files in parallel causes fragmentation of your hard disk. Also, if the connection breaks while transferring all files in parallel, you will have a dozen of useless parts of files. If you do sequential file transfer, you'll have at least some complete files after breakdown.

But the others are right, stability should have priority anyway.

**Comment by yurivict** [comment]
> Transferring multiple files in parallel causes fragmentation of your hard disk.
This is not true, sorry.

> Also, if the connection breaks while transferring all files in parallel, you will have a dozen of useless parts of files. If you do sequential file transfer, you'll have at least some complete files after breakdown.
This should be managed by tox, it should be continuing as much as possible.

Please make concurrent file transfers work.

**Comment by yurivict** [comment]
https://github.com/irungentoo/toxcore/issues/1621
